### ISSUE TITLE
Should ApplyChangesKind have a member for submissions?

### ISSUE THREAD

**Issue by amcasey** [dotnet/roslyn#4972]
Since ```ChangeDocument``` is workspace-wide, it is too general to be allowed by ```InteractiveWorkspace```.  Maybe there should be a ```ChangeActiveSubmission``` member?

**Comment by amcasey** [comment]
FYI @dpoeschl

**Comment by amcasey** [comment]
Thoughts @DustinCampbell @Pilchie @jasonmalinowski?

**Comment by jasonmalinowski** [comment]
Rather than calling out "active submissions" specifically, should we just have a "can I change this document" API?

**Comment by Pilchie** [comment]
Tag @mattwar

**Comment by amcasey** [comment]
This seems especially relevant in VB because it looks like the pretty printer checks for ```ChangeDocument```.

**Comment by amcasey** [comment]
Based on offline discussion, it sounds like we're leaning towards a new API that asks whether a specific document is modifiable.

**Comment by amcasey** [comment]
It seems like ```IDocumentSupportsSuggestionService``` already provides this functionality.

Please re-open if you disagree.

**Comment by jasonmalinowski** [comment]
Reopening since I disagree: it's internal, and thus doesn't solve the problem of extenders.

**Comment by mattwar** [comment]
Yes, add an API that asks about specific documents.   CanChangeDocument(DocumentId)

**Comment by CyrusNajmabadi** [comment]
Not planning on making any changes here.
